def test_old_testresult(self):


    class Test(unittest2.TestCase):

        def testSkip(self):
            self.skipTest('foobar')

        @unittest2.expectedFailure
        def testExpectedFail(self):
            raise TypeError

        @unittest2.expectedFailure
        def testUnexpectedSuccess(self):
            pass
    for (test_name, should_pass) in (('testSkip', True), ('testExpectedFail', True), ('testUnexpectedSuccess', False)):
        test = Test(test_name)
        self.assertOldResultWarning(test, int((not should_pass)))
